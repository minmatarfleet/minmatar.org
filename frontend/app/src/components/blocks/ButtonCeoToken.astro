---
import { i18n } from '@helpers/i18n'
const { t, translatePath } = i18n(Astro.url)

import type {  ButtonSizes, ButtonColors, CorporationObject } from '@dtypes/layout_components';

interface Props {
    corporation:    CorporationObject;
    color?:         ButtonColors;
    size?:          ButtonSizes;
    narrow?:        boolean;
    dialog_title:   string;
    redirect?:      string;
}

const {
    corporation,
    color = 'fleet-red',
    size = 'lg',
    narrow = false,
    dialog_title,
    redirect = Astro.url,
} = Astro.props;

import { query_string } from '@helpers/string';
const ADD_CEO_TOKEN_INIT_PARTIAL_URL = `${translatePath('/redirects/add_ceo_token_init')}?redirect=${redirect}`

interface Message {
    message: string
}

type Params = CorporationObject & Message

const params:Params = {
    alliance_id: corporation.alliance_id,
    corporation_id: corporation.corporation_id,
    alliance_name: corporation.alliance_name,
    corporation_name: corporation.corporation_name,
    corporation_type: corporation.corporation_type,
    active: corporation.active,
    biography: corporation.biography,
    introduction: corporation.introduction,
    requirements: corporation.requirements,
    timezones: corporation.timezones,
    message: t('add_ceo_token_disclaimer'),
}

import Button from '@components/blocks/Button.astro';
---

<Button
    type="button"
    color={color}
    size={size}
    narrow={narrow}
    href={ADD_CEO_TOKEN_INIT_PARTIAL_URL}
    x-data={`{
        add_ceo_token(accepted) {
            if (accepted) window.location.href = "${ADD_CEO_TOKEN_INIT_PARTIAL_URL}"
        },
        show_add_ceo_disclaimer() {
            show_confirm_dialog({
                title: '${dialog_title}',
                partial: '${translatePath('/partials/dialog_with_corporation/')}?${query_string(params)}',
            }).then( (accepted) => this.add_ceo_token(accepted) )
        }
    }`}
    x-on:click.prevent="show_add_ceo_disclaimer"
>
    <slot />
</Button>