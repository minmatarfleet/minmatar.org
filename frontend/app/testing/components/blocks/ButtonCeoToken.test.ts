import { experimental_AstroContainer as AstroContainer } from "astro/container";
import { expect, test } from "vitest";
import ButtonCeoToken from "@components/blocks/ButtonCeoToken.astro";

test("ButtonCeoToken defaults", async () => {
  const container = await AstroContainer.create();
  const result = await container.renderToString(ButtonCeoToken, {
    props: {
      corporation: {
        alliance_id: 12,
        coporation_id: 32,
        alliance_name: "Chickens R Us",
        corporation_name: "Top Cluck",
        corporation_type: "feathered",
        biography: "lorem ipsum chickensum",
        introduction: "in a world... with chickens",
        requirements: "must be a chicken",
        timezones: "CHICKUTC",
      },
    },
  });

  expect(result).toMatchSnapshot();
});
