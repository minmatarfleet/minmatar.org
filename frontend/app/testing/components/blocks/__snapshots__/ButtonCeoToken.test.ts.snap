// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ButtonCeoToken defaults 1`] = `
"<a href="/redirects/add_ceo_token_init?redirect=https://example.com/" type="button" x-data="{
        add_ceo_token(accepted) {
            if (accepted) window.location.href = &#34;/redirects/add_ceo_token_init?redirect=https://example.com/&#34;
        },
        show_add_ceo_disclaimer() {
            show_confirm_dialog({
                title: 'undefined',
                partial: '/partials/dialog_with_corporation/?alliance_id=12&#38;corporation_id=undefined&#38;alliance_name=Chickens+R+Us&#38;corporation_name=Top+Cluck&#38;corporation_type=feathered&#38;active=undefined&#38;biography=lorem+ipsum+chickensum&#38;introduction=in+a+world...+with+chickens&#38;requirements=must+be+a+chicken&#38;timezones=CHICKUTC&#38;message=Corporation+applications+are+currently+not+available+until+a+CEO+token+is+added.+If+you+are+not+one+of+the+CEOs+of+this+corporation%2C+you+may+close+this+dialog.%0A%0AWould+you+like+to+proceed+with+adding+the+CEO+token%3F',
            }).then( (accepted) => this.add_ceo_token(accepted) )
        }
    }" x-on:click.prevent="show_add_ceo_disclaimer" data-size="lg" style="--button_background_color: var(--fleet-red); --button_color: var(--fleet-yellow)" data-astro-cid-r5e2wo56="true" class="[ button ]"> <div class="flex-inline" data-astro-cid-r5e2wo56="true" data-astro-cid-ud5fqf5k style="--flex_inline_gap: var(--space-xs);--flex_inline_justification: center;">       </div>   </a> "
`;
